### PR TITLE
Refactor contracts module

### PR DESCRIPTION
**Pull request by nachfq**
## 1. Centralized Configuration

✅ **Risk multiplier constants moved to dedicated constants** - Removed hardcoding

`src/contracts/constants/risk-multipliers.constants.ts` (new)

`src/contracts/constants/index.ts` (new)

`src/contracts/services/cache-statistics.service.ts` (modified)

`src/contracts/services/contract-bid-assessment.service.ts` (modified)

## 2. Enhanced Validation

✅ **UUID validation for blockchain IDs** - Type-safe DTOs with proper validation

`src/contracts/dto/contract-query.dto.ts` (new)

`src/contracts/dto/suggested-bids-query.dto.ts` (new)

`src/contracts/dto/create-contract.dto.ts` (deleted)

`src/contracts/dto/update-contract.dto.ts` (deleted)

`src/user-contracts/dto/create-user-contract.dto.ts` (modified)

## 3. Modular Service Architecture

✅ **Separation of concerns** - Individual services for specific responsibilities

`src/contracts/services/contract-bid-calculator.service.ts` (new)

`src/contracts/services/contract-bid-assessment.service.ts` (new)

`src/contracts/services/contract-history.service.ts` (new)

`src/contracts/services/contract-enrichment.service.ts` (new)

`src/contracts/services/cache-statistics.service.ts` (new)

`src/contracts/services/contract-query-builder.service.ts` (new)

`src/contracts/contracts.utils.service.ts` (deleted)

## 4. Centralized Error Handling

✅ **Unified error system** - Specific codes and helper functions

`src/contracts/contracts.errors.ts` (new)

`src/contracts/contracts.controller.ts` (modified)

`src/contracts/contracts.service.ts` (modified)

## 5. Type Safety

✅ **Enhanced interfaces and type guards** - Comprehensive contract interfaces

`src/contracts/interfaces/contract.interfaces.ts` (new)

`src/contracts/contracts.service.ts` (modified)

`src/alerts/alert-monitoring.service.ts` (modified)

## 6. Code Structure

✅ **Barrel exports** - Cleaner imports and better organization

`src/contracts/dto/index.ts` (new)

`src/contracts/services/index.ts` (new)

`src/contracts/contracts.controller.ts` (modified)

`src/contracts/contracts.module.ts` (modified)

## 7. Standardized Logging

✅ **Consistent naming** - Logger(ClassName.name) and appropriate levels

`src/contracts/contracts.service.ts` (modified)

`src/contracts/contracts.controller.ts` (modified)

## 8. Comprehensive Unit Testing

✅ **Complete test coverage** - Individual test suites for all new services

`src/contracts/services/contract-bid-calculator.service.spec.ts` - Bid calculation logic, risk multipliers, edge cases

`src/contracts/services/contract-bid-assessment.service.spec.ts` - Risk assessment algorithms, cache statistics integration

`src/contracts/services/contract-history.service.spec.ts` - Bidding history retrieval, data formatting

`src/contracts/services/contract-enrichment.service.spec.ts` - Contract data enrichment, calculated fields

`src/contracts/services/cache-statistics.service.spec.ts` - Cache utilization metrics, performance tracking

`src/contracts/contracts.service.spec.ts` (modified)

`src/auth/auth.service.spec.ts` (modified) - Updated for test consistency

**Testing Infrastructure** - Jest setup file, logging utilities, mocked dependencies

`jest.setup.ts` (new)

`eslint.config.mjs` (modified)

`package.json` (modified)

## 9. Complete Module Documentation

✅ **Comprehensive documentation** - Detailed module specifications and usage examples

`src/contracts/docs/contracts.md` (new) - Complete module overview, API documentation, usage examples

`src/contracts/docs/eviction-risk.md` (new) - Detailed eviction risk calculation methodology

`src/contracts/contracts.md` → `src/contracts/docs/contracts.md` (moved)

`src/contracts/eviction-risk.md` → `src/contracts/docs/eviction-risk.md` (moved)

### CODE REVIEW

File: src/contracts/interfaces/contract.interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
+/**
+ * Shared types and interfaces for contract services
+ */
+
+import { Contract } from '../entities/contract.entity';
+
+// TODOREFACTOR: Import this interface from the common ABI. Should not be defined here.
```
Comment (augustocollerone) [inline src/contracts/interfaces/contract.interfaces.ts:7]: Deberiamos implementar este TODOREFACTOR ahora?

Comment (ifqbuilder) [reply src/contracts/interfaces/contract.interfaces.ts:7]: Noted!
